snapcraft/lifecycle.py

```diff
@@ -371,3 +356,73 @@ def _validate_relative_paths(files):
     for d in files:
         if os.path.isabs(d):
             raise PluginError('path "{}" must be relative'.format(d))
+
+
+def _add_required_schema(plugin_schema):
+    if plugin_schema:
+        schema = plugin_schema
+        if 'properties' not in schema:
+            schema['properties'] = {}
+    else:
+        schema = {
```

Comment:
why don't you load `schema/snapcraft.yaml` which is already loaded in `yaml.py`, so you can just pass the relevant schema part for properties defined in https://github.com/ubuntu-core/snapcraft/blob/master/schema/snapcraft.yaml#L97 and also add the missing `after`, `organize`, ... , there.